### PR TITLE
Fix rename change type & support 'change in type'

### LINKED ISSUE #563: rename file got diff with change_type: R100
when I renamed file A to B like this
```bash
$ git mv A B
$ git status 
On branch master
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

	modified:   .gitmodules
	renamed:    A -> B
	modified:   Cprogramming
```
but using gitpython in ipython3 I got this:
```python
In [135]: repo.commit('HEAD~2').diff(repo.head.commit)
Out[135]: 
[<git.diff.Diff at 0x7fc52ab13730>,
 <git.diff.Diff at 0x7fc529943c80>,
 <git.diff.Diff at 0x7fc529943d90>,
 <git.diff.Diff at 0x7fc52994ec80>]

In [136]: repo.commit('HEAD~2').diff(repo.head.commit)[2].b_path
Out[136]: 'Cprogramming'

In [137]: repo.commit('HEAD~2').diff(repo.head.commit)[1].b_path
Out[137]: 'B'

In [138]: repo.commit('HEAD~2').diff(repo.head.commit)[1].change_type
Out[138]: 'R100'
```

is that normal or something went wrong?

### PR DESCRIPTION
**Pull request by LeResKP**
Hello,

It fixes #563, we now parse correctly the change type and store the score if given. I didn't manage to create a copied status in my git so I didn't support it for now. Hope I will find some time later to give another try.

Regards,
Aurélien

### CODE REVIEW

File: git/diff.py [hunk 1]
```diff
@@ -471,6 +475,11 @@ def handle_diff_line(line):
 
             meta, _, path = line[1:].partition('\t')
             old_mode, new_mode, a_blob_id, b_blob_id, change_type = meta.split(None, 4)
+            # Change type can be R100
+            # R: status letter
+            # 100: score (in case of copy and rename)
+            change_type = change_type[0]
```
Comment (Byron) [inline git/diff.py]: This one is interesting: change_type is reassigned, and thus just a single character. One line later, change_type is treated as the original string to get the score, which probably is always empty now.

File: git/test/test_diff.py [hunk 2]
```diff
@@ -108,8 +108,36 @@ def test_diff_with_rename(self):
...
+        assert isinstance(str(diff), str)
+
+        output = StringProcessAdapter(fixture('diff_change_in_type_raw'))
+        diffs = Diff._index_from_raw_format(self.rorepo, output)
+        self.assertEqual(len(diffs), 1)
+        diff = diffs[0]
+        self.assertEqual(diff.rename_from, None)
+        self.assertEqual(diff.rename_to, None)
+        self.assertEqual(diff.change_type, 'T')
+        self.assertEqual(len(list(diffs.iter_change_type('T'))), 1)
```
Comment (Byron) [inline git/test/test_diff.py]: I think a test for `score` is missing. It's likely to not work in the current implementation.

**Comment by LeResKP** [comment]
oh sorry, I didn't red this old commit correctly, it was not finished.... I fixed it.

**Comment by Thrimbda** [comment]
glad to see this issue solved

**Pull request merged**